rdkit/Chem/Fraggle/FraggleSim.py

```diff
@@ -29,356 +29,285 @@
-#keep correct if >60% query mol
+# smarts used to find appropriate fragment for
+# would use SMARTS: [$([#0][r].[r][#0]),$([#0][r][#0])]
+# but RDkit doesn't support component SMARTS in recursive one - $([#0][r].[r][#0])
+# hence split into two
+cSma1 = Chem.MolFromSmarts("[#0][r].[r][#0]")
+cSma2 = Chem.MolFromSmarts("[#0][r][#0]")
 
 
-#start
 def delete_bonds(mol, bonds, ftype, hac):
+  """ Fragment molecule on bonds and reduce to fraggle fragmentation SMILES.
+  If none exists, returns None """
 
-  #use the same parent mol object and create editable mol
+  # Replace the given bonds with attachment points (B1-B2 -> B1-[*].[*]-B2)
+  # print("Breaking bonds between atoms: ", bonds)
   em = Chem.EditableMol(mol)
-
-  #loop through the bonds to delete
-  #print "Breaking bonds between atoms: ",bonds
-
-  for b in bonds:
-    #remove the bond
-    em.RemoveBond(b[0], b[1])
-
-    #now add attachement points
-    newAtomA = em.AddAtom(Chem.Atom(0))
-    em.AddBond(b[0], newAtomA, Chem.BondType.SINGLE)
-
-    newAtomB = em.AddAtom(Chem.Atom(0))
-    em.AddBond(b[1], newAtomB, Chem.BondType.SINGLE)
-
-  #should be able to get away without sanitising mol
-  #as the valencies should be okay
+  for bond in bonds:
+    em.RemoveBond(bond[0], bond[1])
+    em.AddBond(bond[0], em.AddAtom(Chem.Atom(0)), Chem.BondType.SINGLE)
+    em.AddBond(bond[1], em.AddAtom(Chem.Atom(0)), Chem.BondType.SINGLE)
   modifiedMol = em.GetMol()
 
-  #do not do a full sanitization, but do find rings and calculate valences:
+  # should be able to get away without sanitising mol as the valencies should be okay
+  # do not do a full sanitization, but do find rings and calculate valences:
   Chem.SanitizeMol(modifiedMol, Chem.SanitizeFlags.SANITIZE_PROPERTIES |
                    Chem.SanitizeFlags.SANITIZE_SYMMRINGS)
 
   fragmented_smi = Chem.MolToSmiles(modifiedMol, True)
-
-  #print fragmented_smi
-  fraggle_framentation = select_fragments(fragmented_smi, ftype, hac)
-
-  return fraggle_framentation
+  return select_fragments(fragmented_smi, ftype, hac)
 
 
 def is_ring_cut_valid(smi):
-  #to check is a fragment is a valid ring cut, it needs to match the
-  #smarts: [$([#0][r].[r][#0]),$([#0][r][#0])]
-  atom_count = 0
-  valid = False
-
+  """ to check is a fragment is a valid ring cut, it needs to match the
+  SMARTS: [$([#0][r].[r][#0]),$([#0][r][#0])] """
+  # We suppress rdApp errors temporarily to avoid the "non-ring atom marked aromatic errors"
+  # There is no way to get the current status, so we assume that to be enabled.
+  rdBase.DisableLog('rdApp.error')
   m = Chem.MolFromSmiles(smi)
-  if m is not None:
-    #use gobal smarts
-    if (m.HasSubstructMatch(cSma1) or m.HasSubstructMatch(cSma2)):
-      atom_count = m.GetNumAtoms()
-      valid = True
-
-  return valid, atom_count
+  rdBase.EnableLog('rdApp.error')
+  if m is not None and (m.HasSubstructMatch(cSma1) or m.HasSubstructMatch(cSma2)):
+    return True, m.GetNumAtoms()
+  else:
+    return False, 0
 
```

Comment:
I changed the code. The initial intention was probably to return a flag (is_valid_ringcut) and the number of atoms was an after thought. I changed this now so that the function only does what the name implies and move the calculation of number of atoms to the calling code.